drivers/pcie/host/msi.c

```diff
@@ -171,7 +172,11 @@ uint8_t pcie_msi_vectors_allocate(pcie_bdf_t bdf,
 		vectors[req_vectors].bdf = bdf;
 	}
 
+#ifdef CONFIG_PCIE_CONTROLLER
+	return pcie_ctrl_msi_vectors_allocate(priority, vectors, n_vector);
+#else
```

Comment:
You are already exposing a dedicated msi_vector (generic_msi_vector) as arch_msi_vector_t... shouldn't you consider exposing the arch_pcie_msi_vectors_* functions as well then? 
because now, arch that do are exposing a pcie controller, the generic one, are always going to use the above msi_vector structure anyway.

Reply:
I can't remeber why I did that, let me try again

Reply:
OK, well thanks, I removed an useless commit !